x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings/latest_findings_container.tsx

```diff
@@ -68,14 +69,22 @@ export const LatestFindingsContainer = ({ dataView }: FindingsBaseProps) => {
    */
   const findingsGroupByNone = useLatestFindings({
     ...getPaginationQuery({
-      pageIndex: urlQuery.pageIndex,
-      pageSize: pageSize || urlQuery.pageSize,
+      pageIndex: 0,
+      pageSize: MAX_ITEMS,
     }),
     query: baseEsQuery.query,
     sort: urlQuery.sort,
     enabled: !baseEsQuery.error,
   });
 
+  const slicedPage = useMemo(() => {
+    const pageSizes = pageSize !== undefined ? pageSize : 0;
```

Comment:
checking for `undefined` here is a bit  unfortunate, as `pageSize` can never be `0` as it is set initially by `urlQuery` which has a default value for it, but `useLocalStorage` returns `T | undefined` 

i also noticed we call both `setPageSize(page.size)`  and `setUrlQuery(page.size)`

it's out-of-scope for this PR, but both of these issues are because we now have 2 sources of truth (kinda) - local storage / url. we should consolidate them to a single state object that always has the type of `ReturnType<typeof getDefaultQuery>`.

Reply:
You may want to give this another review, as I've refactored a bit and created some hooks to avoid duplicate code.

Reply:
Ricky is on vacation this week, so I'll be finishing this PR up.